sample-apps/ruby-rails-sample-app/Gemfile

```diff
@@ -0,0 +1,34 @@
+source "https://rubygems.org"
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+ruby "2.7.5"
+
+# Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
+gem "rails", "~> 7.0.3"
+
+# Use the Puma web server [https://github.com/puma/puma]
+gem "puma", "~> 5.0"
+
+# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
+gem "tzinfo-data", platforms: %i[ mingw mswin x64_mingw jruby ]
+
+group :development, :test do
+  # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
+  gem "debug", platforms: %i[ mri mingw x64_mingw ]
+end
+
+group :development do
+  # Speed up commands on slow machines / big apps [https://github.com/rails/spring]
+  # gem "spring"
+end
+
+gem 'aws-sdk-s3', '~> 1'
+gem 'faraday'
+
+gem 'opentelemetry-sdk'
+
+gem 'opentelemetry-propagator-xray'
+
+gem 'opentelemetry-sdk'
+gem 'opentelemetry-exporter-otlp'
+gem 'opentelemetry-instrumentation-all'
```

Comment:
Instead of depending on `-all` I suggest that you add dependencies to only libraries that you are going to instrument. You can use this as reference https://github.com/aws-observability/aws-otel-ruby/blob/main/sample-apps/manual-instrumentation/ruby-on-rails/Gemfile#L28

If you look into `Gemfile.lock` this is bringing a lot of unnecessary dependencies. E.g.: opentelemetry-instrumentation-faraday, opentelemetry-instrumentation-mongo

Reply:
I cleaned up the dependencies to only use what is needed